### PR TITLE
Add build and run commands for examples

### PR DESCRIPTION
**Pull request by muja**

### CODE REVIEW

File: lib/cargo.js [hunk 1]
```diff
@@ -108,6 +108,21 @@ export function provideBuilder() {
           args: [ 'update' ].concat(args),
           sh: false,
           errorMatch: []
+        },
+        {
+          name: `Cargo: build example`,
+          exec: cargoPath,
+          args: [ 'build', '--example', '{FILE_ACTIVE_NAME_BASE}' ].concat(args),
```
Comment (colin-kiegel) [inline lib/cargo.js:115]: How is `{FILE_ACTIVE_NAME_BASE}` being replaced with anything?

**Comment by colin-kiegel** [comment]
A general thought is: Since cargo supports a lot of such special parameter combinations and plugins with even more parameters, what will be the borderline for commands to be included into the main package?

Atom-build already supports to add custom build commands on a per-project basis. Maybe it would help rust-development to have custom build commands that are global to all projects but _specific_ for rust?

Maybe something like [this](https://github.com/noseglid/atom-build#specifying-a-custom-build-command) but for the whole atom-build-cargo package. It would be nice if these custom commands could reference the built-in regexps without complicated new syntax or great knowledge about internals of this package.

On the other hand this flexibility might lead to less people contributing to the atom-build-cargo package. It's just the command list getting longer and longer right now.

**Comment by noseglid** [comment]
I agree @colin-kiegel . A better solution for all these permutations will soon be required.
I'm not very familiar with rust and cargo myself, but the most viable long term solution, I think, would be to in some way extract the available targets from cargo itself. This is how for instance [atom-build-gradle](https://github.com/AtomBuild/atom-build-gradle/blob/master/lib/gradle.js) does it. Would that be possible?

**Comment by colin-kiegel** [comment]
I can't see a good way to automatically reason about available and meaningful cargo build commands. I think it's more about individual workflows, how often you use cargo to produce html-documentation, run benchmarks, or run examples. Cargo itself tries to be a swiss knife (with plugins). It is already more than a build tool (you can install binary packages for example). And cargo-plugins may also introduce very different behaviour

File: lib/cargo.js (hunk 1 above)
Comment (colin-kiegel) [inline lib/cargo.js:115]: I just realized [these replacement rules](https://github.com/noseglid/atom-build#replacements) I was not aware of.

Comment (noseglid) [inline lib/cargo.js:115]: Yes, that'll work.

**Comment by noseglid** [comment]
@colin-kiegel I agree we should think of a different solution than iterating all permutations as separate targets, but that is really out of the scope of this PR.

@muja Thanks for this!

**Comment by muja** [comment]
@colin-kiegel @noseglid to add to the discussion, I was trying to find a generic/clever solution for the example case but all I could find unfortunately were the given string replacements, so this was the best i could come up with.

My idea was something along the lines of: a task for "Build current file" and look on every build where the file is located. If under `/example`, build with "--example", if under `/src`, build whole project. if under `/test`, build test etc. Or maybe a "Build all" that builds the project + all examples + tests.

**Pull request merged**